File: java/socketd/src/main/java/org/noear/socketd/transport/core/ChannelSupporter.java
package org.noear.socketd.transport.core;

/**
 * 通道支持者（创建通道的参数）
 *
 * @author noear
 * @since 2.1
 */
public interface ChannelSupporter<S> {
    /**
     * 处理器
     */
    Processor getProcessor();

    /**
     * 配置
     */
    Config getConfig();

    /**
     * 通道助理
     */
    ChannelAssistant<S> getAssistant();
}

File: swift/socketd/socketd/src/socketd/transport/core/ChannelSupporter.swift
//
//  ChannelSupporter.swift
//  socketd
//
//  Created by noear on 2023/12/27.
//

import Foundation

/**
 * 通道支持者（创建通道的参数）
 *
 * @author noear
 * @since 2.1
 */

protocol ChannelSupporter{
    /**
     * 处理器
     */
    func getProcessor() -> Processor;
    
    /**
     * 配置
     */
    func getConfig() -> Config;
    
    /**
     * 通道助理
     */
    func getAssistant() -> ChannelAssistant;
}
